### PR TITLE
JIT: Skip cost improvement check in 3-opt for exceptionally costly layouts

### LINKED ISSUE #109984: JIT: Assertion failed '(newLayoutCost < currLayoutCost) || Compiler::fgProfileWeightsEqual(newLayoutCost, currLayoutCost, 0.001)' during 'Optimize layout'
Hitting in `libraries-jitstress` again ([example run](https://dev.azure.com/dnceng-public/public/_build/results?buildId=872721&view=results)). Looks like #109835 didn't completely fix #109831.

cc @dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Fixes #109984.

### CODE REVIEW

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -4927,21 +4927,33 @@ Compiler::ThreeOptLayout::ThreeOptLayout(Compiler* comp)
 //   endPos - The inclusive ending index of the region
 //
 // Returns:
-//   The region's layout cost
+//   The region's layout cost, or 'BB_MAX_WEIGHT' for pathalogically costly layouts
```
Comment (teo-tsirpanis) [inline src/coreclr/jit/fgopt.cpp:4930]:
```suggestion
//   The region's layout cost, or 'BB_MAX_WEIGHT' for pathologically costly layouts
```

**Review by mslovelymia — COMMENTED** [review]
dont wanna approve anything

**Comment by amanasifkhalid** [comment]
cc @dotnet/jit-contrib, @AndyAyersMS PTAL. `libraries-jitstress` failures don't look related. For the cases I looked at, the failure in #109984 only seems to hit under `JitOptRepeat` stress scenarios where `optSetBlockWeights` inflates loop block weights, thus causing the layout cost calculation to lose precision when summing large values. The simplest solution seems to be to skip the assertion if we detect that we're dealing with some pathologically large layout cost.

**Review by AndyAyersMS — APPROVED** [review]

**Pull request merged**